src/common/NetworkManager.ts

```diff
@@ -218,8 +217,17 @@ export class NetworkManager extends EventEmitter {
     });
   }
 
-  async setUserAgent(userAgent: string): Promise<void> {
-    await this._client.send('Network.setUserAgentOverride', { userAgent });
+  async setUserAgent(
+    userAgent: string,
+    userAgentMetadata?: Protocol.Emulation.UserAgentMetadata
+  ): Promise<void> {
+    const params = { userAgent: userAgent };
```

Comment:
`await this._client.send('Network.setUserAgentOverride', { userAgent, userAgentMetadata });` should work AFAIK because undefined won't be serialised to JSON.

Reply:
Ah, nice, will try this.